src/appfl/run_grpc_client.py

```diff
@@ -18,46 +18,55 @@
 from .protos.client import FLClient
 from .misc.data import Dataset
 
+
 def update_model_state(comm, model, round_number):
     new_state = {}
     for name in model.state_dict():
         nparray = comm.get_tensor_record(name, round_number)
         new_state[name] = torch.tensor(nparray)
     model.load_state_dict(new_state)
 
-def run_client(cfg        : DictConfig,
-               comm_rank  : int,
-               model      : nn.Module,
-               train_data : Dataset) -> None:
+
+def run_client(
+    cfg: DictConfig, cid: int, model: nn.Module, train_data: Dataset, gpu_id : int = 0
+) -> None:
     """Launch gRPC client to connect to the server specified in the configuration.
 
     Args:
         cfg (DictConfig): the configuration for this run
-        comm_rank (int): MPI rank
+        cid (int): cliend_id  
         model (nn.Module): neural network model to train
         train_data (Dataset): training data
+        gpu_id (int): MPI rank  
```

Comment:
Why is this "MPI rank"? Isn't this just GPU ID on a machine? And, ideally, gRPC should be independent to MPI.

Reply:
revised